crypto/txscript/src/data_stack.rs

```diff
@@ -111,102 +146,44 @@ fn deserialize_i64(v: &[u8]) -> Result<i64, TxScriptError> {
-        self.0.partial_cmp(other)
-    }
-}
-
-impl Deref for Kip10I64 {
-    type Target = i64;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl OpcodeData<Kip10I64> for Vec<u8> {
-    #[inline]
-    fn deserialize(&self) -> Result<Kip10I64, TxScriptError> {
-        match self.len() > DEFAULT_SCRIPT_NUM_LEN_KIP10 {
-            true => Err(TxScriptError::NumberTooBig(format!(
-                "numeric value encoded as {:x?} is {} bytes which exceeds the max allowed of {}",
-                self,
-                self.len(),
-                DEFAULT_SCRIPT_NUM_LEN_KIP10
-            ))),
-            false => deserialize_i64(self).map(Kip10I64),
-        }
-    }
-
-    #[inline]
-    fn serialize(from: &Kip10I64) -> Result<Self, SerializationError> {
-        if from.0 == i64::MIN {
-            return Err(SerializationError::NumberTooLong(from.0));
-        }
-        Ok(serialize_i64(&from.0))
-    }
-}
+// TODO: Rename to DefaultSizedEncodeInt when KIP-10 is activated
+pub type Kip10I64 = SizedEncodeInt<8>;
 
 impl OpcodeData<i64> for Vec<u8> {
     #[inline]
     fn deserialize(&self) -> Result<i64, TxScriptError> {
-        match self.len() > DEFAULT_SCRIPT_NUM_LEN {
-            true => Err(TxScriptError::NumberTooBig(format!(
-                "numeric value encoded as {:x?} is {} bytes which exceeds the max allowed of {}",
-                self,
-                self.len(),
-                DEFAULT_SCRIPT_NUM_LEN
-            ))),
-            false => deserialize_i64(self),
-        }
+        // TODO: Change LEN to 8 once KIP-10 is activated
+        OpcodeData::<SizedEncodeInt<4>>::deserialize(self).map(i64::from)
     }
 
     #[inline]
     fn serialize(from: &i64) -> Result<Self, SerializationError> {
-        if from == &i64::MIN {
-            return Err(SerializationError::NumberTooLong(*from));
-        }
-        Ok(serialize_i64(from))
+        // Note that serialization and deserialization use different LEN.
+        // This is because prior to KIP-10, only deserialization size was limited.
+        // It's safe to use 8 here because i32 arithmetic operations (which were the
+        // only ones that were supported prior to KIP-10) can't get to i64::MIN
+        // (the only i64 value that requires more than 8 bytes to serialize).
+        OpcodeData::<SizedEncodeInt<8>>::serialize(&(*from).into())
     }
 }
 
 impl OpcodeData<i32> for Vec<u8> {
     #[inline]
     fn deserialize(&self) -> Result<i32, TxScriptError> {
-        let res = OpcodeData::<i64>::deserialize(self)?;
-        // TODO: Consider getting rid of clamp, since the call to deserialize should return an error
-        // if the number is not in the i32 range (this should be done with proper testing)?
-        Ok(res.clamp(i32::MIN as i64, i32::MAX as i64) as i32)
+        OpcodeData::<SizedEncodeInt<4>>::deserialize(self).map(|v| i64::from(v).try_into().expect("number is within i32 range"))
```

Comment:
Would it be better if tryInto gets implemented on the SizedEncodeInt directly ? Possibly making separate module for the type